src/btree/bt_cursor.c

```diff
@@ -77,8 +74,13 @@ __btcur_bounds_search_near_reposition(WT_SESSION_IMPL *session, WT_CURSOR_BTREE
      * bounds calls. However for the time being we will leave it as is as it is unlikely to present
      * a performance issue.
      */
-    WT_RET(__btcur_bounds_contains_key(
-      session, cursor, &cursor->key, WT_RECNO_OOB, &key_out_of_bounds, &upper));
+    if (CUR2BT(cursor)->type == BTREE_ROW)
+        WT_RET(__btcur_bounds_contains_key(
+          session, cursor, &cursor->key, WT_RECNO_OOB, &key_out_of_bounds, &upper));
+    else
+        WT_RET(__btcur_bounds_contains_key(
+          session, cursor, NULL, cursor->recno, &key_out_of_bounds, &upper));
+
     if (key_out_of_bounds) {
         __wt_cursor_set_raw_key(cursor, upper ? &cursor->upper_bound : &cursor->lower_bound);
```

Comment:
Just confirming, does `__wt_cursor_set_raw_key` set the cursor->recno?

Reply:
Good question!  `__wt_cursor_set_raw_key` calls `__wt_cursor_set_keyv` internally which sets the cursor->recno if it's column store.